src/server.cpp

```diff
@@ -2512,8 +2512,11 @@ void Server::fillMediaCache()
 			{
 				std::string parentpath = m_media.find(filename)->second.path;
 
+				// remove filename and / in front of it from path
+				parentpath = parentpath.substr(0, (parentpath.size()-filename.size())-1);
+
 				// Make sure parent is a parent and not a texture in another mod
-				if (mediapath.find(parentpath.substr(0, parentpath.size()-filename.size())) == 0)
+				if (mediapath.find(parentpath.substr(0, parentpath.find_last_of('/'))) == 0)
```

Comment:
I can see there being a `fs::getBasePathOf(path)`

Reply:
Could probably do that later if someone is in need of it